Sources/Extensions/UIKit/UIEdgeInsetsExtensions.swift

```diff
@@ -44,6 +44,60 @@ extension UIEdgeInsets {
     public init(horizontal: CGFloat, vertical: CGFloat) {
         self.init(top: vertical/2, left: horizontal/2, bottom: vertical/2, right: horizontal/2)
     }
+
+    /// SwifterSwift: Creates an `UIEdgeInsets` based on current value and top offset.
+    ///
+    /// - Parameters:
+    ///   - top: Offset to be applied in to the top edge.
+    /// - Returns: UIEdgeInsets offset with given offset.
+    public func insetBy(top: CGFloat) -> UIEdgeInsets {
+        return UIEdgeInsets(top: self.top + top, left: left, bottom: bottom, right: right)
+    }
+
+    /// SwifterSwift: Creates an `UIEdgeInsets` based on current value and left offset.
+    ///
+    /// - Parameters:
+    ///   - top: Offset to be applied in to the left edge.
+    /// - Returns: UIEdgeInsets offset with given offset.
+    public func insetBy(left: CGFloat) -> UIEdgeInsets {
+        return UIEdgeInsets(top: top, left: self.left + left, bottom: bottom, right: right)
+    }
+
+    /// SwifterSwift: Creates an `UIEdgeInsets` based on current value and bottom offset.
+    ///
+    /// - Parameters:
+    ///   - top: Offset to be applied in to the bottom edge.
+    /// - Returns: UIEdgeInsets offset with given offset.
+    public func insetBy(bottom: CGFloat) -> UIEdgeInsets {
+        return UIEdgeInsets(top: top, left: left, bottom: self.bottom + bottom, right: right)
+    }
+
+    /// SwifterSwift: Creates an `UIEdgeInsets` based on current value and right offset.
+    ///
+    /// - Parameters:
+    ///   - top: Offset to be applied in to the right edge.
+    /// - Returns: UIEdgeInsets offset with given offset.
+    public func insetBy(right: CGFloat) -> UIEdgeInsets {
```

Comment:
Think we can condense all of these into:

```swift
func insetBy(right: CGFloat? = nil, left: CGFloat? = nil, top: CGFloat? = nil, bottom: CGFloat? = nil) {
      // ...
}

func insetBy(vertical: CGFloat? = nil, horizontal: CGFloat? = nil) {
    // ...
}
```
Then users can add the parameters they choose?

Reply:
Hi @SD10 this design will allow the user to call `insets.insetBy()` with no parameters. [Check out our discussion above](https://github.com/SwifterSwift/SwifterSwift/pull/532#discussion_r207216832).

Reply:
Yup, great point again 😆 🥇